From ddf0f49dc949beab8485b3bfb8e88616e4bd8076 Mon Sep 17 00:00:00 2001
From: David Redondo <qt@david-redondo.de>
Date: Fri, 9 Oct 2026 07:14:36 +0000
Subject: [PATCH] Revert "wayland: Report pixelScroll capability for touchpads"

This reverts commit 1f94e2782c071c9c687da1bc0141f55118943623.

Reason for revert: While touchpads indeed have pixel scroll, currently all events
are attributed to this device if it exists also the ones that come from mice. This
has further side effects like slow scrolling in widgets or not working functionality
in QTabbar. If we want to go in this direction it needs more work likely also in
widgets.
Task: QTBUG-150280

Change-Id: I1e774fce9888caa4c485405679b2a45499429813
Reviewed-by: David Edmundson <davidedmundson@kde.org>
(cherry picked from commit 6b7ee05def510c333fbb8944ae669b72a48367f3)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/plugins/platforms/wayland/qwaylandinputdevice.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/plugins/platforms/wayland/qwaylandinputdevice.cpp b/src/plugins/platforms/wayland/qwaylandinputdevice.cpp
index cd15c50fbd0..97385902eca 100644
--- a/src/plugins/platforms/wayland/qwaylandinputdevice.cpp
+++ b/src/plugins/platforms/wayland/qwaylandinputdevice.cpp
@@ -416,13 +416,10 @@ void QWaylandInputDevice::seat_capabilities(uint32_t caps)
 
         auto *pointerGestures = mQDisplay->pointerGestures();
         if (pointerGestures) {
-              QInputDevice::Capabilities  pointerCapabilities = QInputDevice::Capability::Position
-                                                             | QInputDevice::Capability::PixelScroll
-                                                             | QInputDevice::Capability::Scroll;
             // NOTE: The name of the device and its system ID are not exposed on Wayland.
             mTouchPadDevice = new QPointingDevice(
                     QLatin1StringView("touchpad"), 0, QInputDevice::DeviceType::TouchPad,
-                    QPointingDevice::PointerType::Finger, pointerCapabilities,
+                    QPointingDevice::PointerType::Finger, QInputDevice::Capability::Position,
                     MaxTouchPoints, 0, mSeatName, QPointingDeviceUniqueId(), this);
             QWindowSystemInterface::registerInputDevice(mTouchPadDevice);
             mPointerGesturePinch.reset(pointerGestures->createPointerGesturePinch(this));
-- 
GitLab

